### ISSUE TITLE
Evolvution analysis not working for me even on Prody 2.0

### ISSUE THREAD

**Issue by obasekigit** [prody/ProDy#1444]
I tried carrying out similar analysis like in the tutorial and i get an error
![Screenshot from 2022-01-13 20-52-35](https://user-images.githubusercontent.com/84424732/149437880-3ec4d996-553f-4e55-85e8-0ee8189a167e.png)

**Comment by jamesmkrieger** [comment]
It looks like pfam may have changed something on their website. We'll have to look carefully and work out what it is

**Comment by obasekigit** [comment]
Thanks for the response. hoping to see a fix for this problem

**Comment by jamesmkrieger** [comment]
You’re welcome. It’s definitely one we need

**Comment by jamesmkrieger** [comment]
Perhaps it's a security issue with pfam. 

The following code doesn't work:
```
In [1]: import urllib.request
   ...: import ssl

In [2]: url = 'https://pfam.xfam.org/protein/P19491?output=xml'

In [3]: urllib.request.urlopen(url)
---------------------------------------------------------------------------
SSLCertVerificationError                  Traceback (most recent call last)
~/anaconda3/envs/scipion3/lib/python3.8/urllib/request.py in do_open(self, http_class, req, **http_conn_args)
   1353             try:
-> 1354                 h.request(req.get_method(), req.selector, req.data, headers,
   1355                           encode_chunked=req.has_header('Transfer-encoding'))

~/anaconda3/envs/scipion3/lib/python3.8/http/client.py in request(self, method, url, body, headers, encode_chunked)
   1255         """Send a complete request to the server."""
-> 1256         self._send_request(method, url, body, headers, encode_chunked)
   1257 

~/anaconda3/envs/scipion3/lib/python3.8/http/client.py in _send_request(self, method, url, body, headers, encode_chunked)
   1301             body = _encode(body, 'body')
-> 1302         self.endheaders(body, encode_chunked=encode_chunked)
   1303 

~/anaconda3/envs/scipion3/lib/python3.8/http/client.py in endheaders(self, message_body, encode_chunked)
   1250             raise CannotSendHeader()
-> 1251         self._send_output(message_body, encode_chunked=encode_chunked)
   1252 

~/anaconda3/envs/scipion3/lib/python3.8/http/client.py in _send_output(self, message_body, encode_chunked)
   1010         del self._buffer[:]
-> 1011         self.send(msg)
   1012 

~/anaconda3/envs/scipion3/lib/python3.8/http/client.py in send(self, data)
    950             if self.auto_open:
--> 951                 self.connect()
    952             else:

~/anaconda3/envs/scipion3/lib/python3.8/http/client.py in connect(self)
   1424 
-> 1425             self.sock = self._context.wrap_socket(self.sock,
   1426                                                   server_hostname=server_hostname)

~/anaconda3/envs/scipion3/lib/python3.8/ssl.py in wrap_socket(self, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, session)
    499         # ctx._wrap_socket()
--> 500         return self.sslsocket_class._create(
    501             sock=sock,

~/anaconda3/envs/scipion3/lib/python3.8/ssl.py in _create(cls, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, context, session)
   1039                         raise ValueError("do_handshake_on_connect should not be specified for non-blocking sockets")
-> 1040                     self.do_handshake()
   1041             except (OSError, ValueError):

~/anaconda3/envs/scipion3/lib/python3.8/ssl.py in do_handshake(self, block)
   1308                 self.settimeout(None)
-> 1309             self._sslobj.do_handshake()
   1310         finally:

SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1131)

During handling of the above exception, another exception occurred:

URLError                                  Traceback (most recent call last)
<ipython-input-3-1024bad9a91a> in <module>
----> 1 urllib.request.urlopen(url)

~/anaconda3/envs/scipion3/lib/python3.8/urllib/request.py in urlopen(url, data, timeout, cafile, capath, cadefault, context)
    220     else:
    221         opener = _opener
--> 222     return opener.open(url, data, timeout)
    223 
    224 def install_opener(opener):

~/anaconda3/envs/scipion3/lib/python3.8/urllib/request.py in open(self, fullurl, data, timeout)
    523 
    524         sys.audit('urllib.Request', req.full_url, req.data, req.headers, req.get_method())
--> 525         response = self._open(req, data)
    526 
    527         # post-process response

~/anaconda3/envs/scipion3/lib/python3.8/urllib/request.py in _open(self, req, data)
    540 
    541         protocol = req.type
--> 542         result = self._call_chain(self.handle_open, protocol, protocol +
    543                                   '_open', req)
    544         if result:

~/anaconda3/envs/scipion3/lib/python3.8/urllib/request.py in _call_chain(self, chain, kind, meth_name, *args)
    500         for handler in handlers:
    501             func = getattr(handler, meth_name)
--> 502             result = func(*args)
    503             if result is not None:
    504                 return result

~/anaconda3/envs/scipion3/lib/python3.8/urllib/request.py in https_open(self, req)
   1395 
   1396         def https_open(self, req):
-> 1397             return self.do_open(http.client.HTTPSConnection, req,
   1398                 context=self._context, check_hostname=self._check_hostname)
   1399 

~/anaconda3/envs/scipion3/lib/python3.8/urllib/request.py in do_open(self, http_class, req, **http_conn_args)
   1355                           encode_chunked=req.has_header('Transfer-encoding'))
   1356             except OSError as err: # timeout error
-> 1357                 raise URLError(err)
   1358             r = h.getresponse()
   1359         except:

URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1131)>
```

However, the following from https://stackoverflow.com/questions/35569042/ssl-certificate-verify-failed-with-python3 does:
```
In [4]:     scontext = ssl.SSLContext(ssl.PROTOCOL_TLS)
   ...:     scontext.verify_mode = ssl.VerifyMode.CERT_NONE
   ...:     with urllib.request.urlopen(url = url, context=scontext) as f:
   ...:         print(f.read().decode('utf-8'))
   ...: 
<?xml version="1.0" encoding="UTF-8"?>
<!-- information on UniProt entry P19491 (GRIA2_RAT), generated: 12:50:07 17-Feb-2022 -->
<pfam xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xmlns="https://pfam.xfam.org/"
      xsi:schemaLocation="https://pfam.xfam.org/
                          https://pfam.xfam.org/static/documents/schemas/protein.xsd"
      release="35.0"
      release_date="2021-11-04">
  <entry entry_type="sequence" db="uniprot" db_release="2021_03" accession="P19491" id="GRIA2_RAT">
    <description>
<![CDATA[
Glutamate receptor 2 {ECO:0000305}
]]>
    </description>
    <taxonomy tax_id="10116" species_name="Rattus norvegicus (Rat)">Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; Mammalia; Eutheria; Euarchontoglires; Glires; Rodentia; Myomorpha; Muroidea; Muridae; Murinae; Rattus.</taxonomy>
    <sequence length="883" md5="7aacb5aaa0c18097dae56341529b5c3b" crc64="DEFA817027C1CCD1" version="2">MQKIMHISVLLSPVLWGLIFGVSSNSIQIGGLFPRGADQEYSAFRVGMVQFSTSEFRLTPHIDNLEVANSFAVTNAFCSQFSRGVYAIFGFYDKKSVNTITSFCGTLHVSFITPSFPTDGTHPFVIQMRPDLKGALLSLIEYYQWDKFAYLYDSDRGLSTLQAVLDSAAEKKWQVTAINVGNINNDKKDETYRSLFQDLELKKERRVILDCERDKVNDIVDQVITIGKHVKGYHYIIANLGFTDGDLLKIQFGGANVSGFQIVDYDDSLVSKFIERWSTLEEKEYPGAHTATIKYTSALTYDAVQVMTEAFRNLRKQRIEISRRGNAGDCLANPAVPWGQGVEIERALKQVQVEGLSGNIKFDQNGKRINYTINIMELKTNGPRKIGYWSEVDKMVVTLTELPSGNDTSGLENKTVVVTTILESPYVMMKKNHEMLEGNERYEGYCVDLAAEIAKHCGFKYKLTIVGDGKYGARDADTKIWNGMVGELVYGKADIAIAPLTITLVREEVIDFSKPFMSLGISIMIKKPQKSKPGVFSFLDPLAYEIWMCIVFAYIGVSVVLFLVSRFSPYEWHTEEFEDGRETQSSESTNEFGIFNSLWFSLGAFMQQGCDISPRSLSGRIVGGVWWFFTLIIISSYTANLAAFLTVERMVSPIESAEDLSKQTEIAYGTLDSGSTKEFFRRSKIAVFDKMWTYMRSAEPSVFVRTTAEGVARVRKSKGKYAYLLESTMNEYIEQRKPCDTMKVGGNLDSKGYGIATPKGSSLGNAVNLAVLKLNEQGLLDKLKNKWWYDKGECGSGGGDSKEKTSALSLSNVAGVFYILVGGLGLAMLVALIEFCYKSRAEAKRMKVAKNPQNINPSSSQNSQNFATYKEGYNVYGIESVKI</sequence>
    <matches>
      <match accession="PF00060" id="Lig_chan" type="Pfam-A">
        <location start="543" end="824" ali_start="544" ali_end="824" hmm_start="2" hmm_end="149" evalue="2.5e-55" bitscore="197.9" />
      </match>
      <match accession="PF01094" id="ANF_receptor" type="Pfam-A">
        <location start="41" end="381" ali_start="55" ali_end="380" hmm_start="22" hmm_end="353" evalue="1.8e-56" bitscore="203.4" />
      </match>
      <match accession="PF10613" id="Lig_chan-Glu_bd" type="Pfam-A">
        <location start="414" end="529" ali_start="414" ali_end="529" hmm_start="1" hmm_end="115" evalue="4.1e-42" bitscore="154.3" />
      </match>
    </matches>
  </entry>
</pfam>
```

**Comment by jamesmkrieger** [comment]
This still takes a very long time so maybe that's not it.

**Comment by jamesmkrieger** [comment]
The actual fix was to use `request.get(url, verify=False)` and now it works. I'll merge the pull request and then you should be able to use it by updating prody from github.

I'll probably make a release soon too as this seems like an important functionality for users to have fixed.

**Comment by jamesmkrieger** [comment]
This should now be fixed and available in ProDy 2.0.2 that can be obtained from PyPI with pip
